jupyter-book/cellular_structure/integration_environment.yml

```diff
@@ -4,63 +4,30 @@ channels:
   - bioconda
   - defaults
 dependencies:
-  - anndata=0.7.6
-  - bioconductor-singlecellexperiment=1.8.0
+  - anndata=0.8.0
+  - anndata2ri=1.1
   - bbknn=1.5.1
-  - hdf5=1.12.1
-  - ipykernel=6.4.1
-  - ipython=7.28.0
-  - ipython_genutils=0.2.0
-  - ipywidgets=7.6.5
-  - jax=0.3.5
-  - jaxlib=0.3.2
-  - jupyter_client=7.0.4
-  - jupyter_core=4.8.1
-  - jupyter_server=1.11.0
-  - jupyterlab=3.1.14
-  - jupyterlab_pygments=0.1.2
-  - jupyterlab_server=2.8.2
-  - jupyterlab_widgets=1.0.2
-  - matplotlib-base=3.4.3
-  - matplotlib-inline=0.1.3
-  - nbclassic=0.3.2
-  - nbclient=0.5.4
-  - nbconvert=6.2.0
-  - nbformat=5.1.3
-  - notebook=6.4.4
-  - numba=0.55.1
-  - numpyro=0.9.1
-  - pandas=1.2.5
-  - pandoc=2.14.2
-  - pandocfilters=1.5.0
-  - patsy=0.5.2
-  - pip=21.2.4
-  - pynndescent=0.5.4
-  - pyro-api=0.1.2
-  - pyro-ppl=1.7.0
-  - pytables=3.6.1
-  - python=3.8.12
-  - python-annoy=1.17.0
-  - pytorch=1.10.2
-  - pytorch-lightning=1.5.10
-  - r-base=3.6.3
-  - r-seurat=3.2.3
-  - r-spatstat=1.64_1
-  - scanpy=1.8.1
-  - scikit-learn=1.0
-  - scipy=1.7.1
-  - scvi-tools=0.16.4
-  - seaborn=0.11.2
-  - seaborn-base=0.11.2
+  - bioconductor-singlecellexperiment=1.16.0
+  - igraph=0.10.2
+  - jupyterlab=3.5.0
+  - leidenalg=0.9.0
+  - louvain=0.8.0
```

Comment:
Do you need leiden and louvain?!?

Reply:
Good question... **scib** definitely needs one (Louvain I think), I probably added Leiden in case we wanted to do some clustering in the notebook but we don't at the moment so could probably remove it.

Reply:
**scib** needs both so I'm going to leave it here because they will be installed by **pip** anyway

Reply:
Soo if they are hard requirements by scIB they should be installed with scIB. If this is the case I would advise against listing them explicitly here again because it is technically possible that you overwrite what scIB specifies and needs with other hardcoded versions that you specify here.

Reply:
@mumichae recently made leiden default... could make louvain optional in scib then and then remove louvain from this.

Reply:
@LuckyMD I agree, but then Leiden should yet again not be listed explicitly here. None of them should be listed here. Only scib

Reply:
If you don't list them they get installed by pip rather than conda and I have found it's generally more reliable to avoid that if possible. That said, in this case I don't think there were problems so can remove them if you want.

Reply:
True, but I think that my argument from above holds. It makes more sense if scib controls the Leiden version even if its from pip. I'd be happy if you could remove them. Thanks!

Reply:
Just want to throw out that we developed https://scib-metrics.readthedocs.io/en/stable/

which requires neither leiden or louvain as it uses a faster implementation of leiden in python igraph. The metrics are also generally faster (but don't work for bbknn or conos)